### PR TITLE
Feat error404 page

### PR DESCRIPTION
**Pull request by dev129**
Hi @Avdhesh-Varshney 
Closes: #919 

- **Title:** Created 404 Error Landing Page
- **Name:** Devansh Vishwa
- **Identify yourself:** GSSoC Contributor 
### Describe the add-ons or changes you've made 📃
I have created the 404 Error Landing Page. The main file is located at `src/app/components/shared/NotFound.tsx` and imported in `src/app/[projectDomain]/page.tsx`. The page includes the design for the error 404, with relevant text and an option to navigate back to the homepage.

### Screenshots 📷

image

### Type of change ☑️
What sort of change have you made:
- [x] New feature (non-breaking change which adds functionality)

### Checklist: ☑️
- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CONTRIBUTING.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Note to reviewers 📄
Please review the layout and functionality of the 404 Error Landing Page. Let me know if any further changes are needed.

### CODE REVIEW

**Comment by dev129** [comment]
Hi @Avdhesh-Varshney , I did some changes , now please review my PR.

File: package-lock.json
Comment (Avdhesh-Varshney) [inline package-lock.json:1]: .

File: package.json
Comment (Avdhesh-Varshney) [inline package.json:1]: .

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
Logo is not display.
image

Project is present inside the `webmasterlog` folder, why have you created package.json outside of it?

**Comment by dev129** [comment]
Done @Avdhesh-Varshney , please check now.

**Comment by dev129** [comment]
Hi @Avdhesh-Varshney , please check the changes I have made and let me know if anythign else is needed.

**Review by Avdhesh-Varshney — APPROVED** [review]

**Pull request merged**